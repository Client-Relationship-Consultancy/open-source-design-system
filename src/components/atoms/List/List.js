import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

const StyledList = styled.ul`
  color: ${props => props.theme.black.main.hex};
  font-size: 1rem;
  li {
    margin-bottom: 0.5rem;
  }
`

StyledList.displayName = "StyledList"

const List = props => {
  return <StyledList as={props.listType}>{props.children}</StyledList>
}

export default List

List.defaultProps = {
  listType: "ul",
}

List.propTypes = {
  listType: PropTypes.oneOf(["ul", "ol"]),
  children: PropTypes.node,
}
