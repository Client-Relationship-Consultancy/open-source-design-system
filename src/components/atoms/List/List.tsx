import React from "react";
import styled from "styled-components";
import { colourPalette } from "../../../brandColours";

const StyledList = styled.ul`
  color: ${props => props.theme.black.main.hex};
  font-size: 1rem;
  li {
    margin-bottom: 0.5rem;
  }
`;

StyledList.defaultProps = {
  theme: colourPalette.examplePalette,
};

StyledList.displayName = "StyledList";

type HTMLListProps = React.HTMLProps<HTMLOListElement> & React.HTMLProps<HTMLUListElement>;

interface Props extends HTMLListProps {
  listType?: "ul" | "ol";
  children?: React.ReactNode;
}

const List: React.FC<Props> = ({ listType = "ul", children }) => {
  return <StyledList as={listType}>{children}</StyledList>;
};

export default List;
