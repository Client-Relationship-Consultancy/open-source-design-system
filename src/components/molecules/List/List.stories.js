import React from "react"
import { storiesOf } from "@storybook/react"
import { boolean } from "@storybook/addon-knobs"
import { ListItem, ListContainer } from "index"

const stories = storiesOf("List", module)

const array = ["Item 1", "Item 2", "Item 3", "Item 4"]

const renderArray = arr =>
  arr.map(item => (
    <ListItem key={item} padding={boolean("Default Padding on items", true)}>
      {item}
    </ListItem>
  ))

stories.add("Standard", context => {
  const text = {
    description: [
      "This is a standard List component. It is made of two componnents, 'ListContainer' and 'ListItem'",
    ],
    dos: [
      "Use this to display a simple list of items",
      "Use the `border` prop when there is no immediate container for this component",
      "The line between each list item should be full width and touches both ends of the surrounding container",
      "Make sure the content inside the list all align if the default padding is off",
    ],
    donts: [
      "Do not use this list to replace bullet points",
      "Do not have a trailing end line if there is no space between the last list item and the container border",
    ],
  }

  Object.assign(context, text)

  return (
    <ListContainer endingLine={boolean("Ending Line", false)} border>
      {renderArray(array)}
    </ListContainer>
  )
})
