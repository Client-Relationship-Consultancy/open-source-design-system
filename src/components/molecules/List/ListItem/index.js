import ListItem, { IListItem } from "./ListItem"

export { IListItem }
export default ListItem
