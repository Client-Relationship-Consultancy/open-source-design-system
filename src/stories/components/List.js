import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"
import { colourPalette } from "../../brandColours"

const { black } = colourPalette.examplePalette

const ListContainer = styled.div`
  margin: 0 0 2rem;
  > h4 {
    font-size: 1rem;
    margin: 0 0 0.5rem;
    color: ${black.main.hex};
    width: ${props => props.width};
  }
`

const StyledList = styled.ul`
  margin: 0;
  padding: 0 0 0 2rem;
  > li {
    margin: 0 0 0.5rem;
    color: ${black.main.hex};
  }
`

const List = props => {
  const { heading, list, width } = props
  // Iterates through array and renders list of li
  const renderList = li => li.map((text, index) => <li key={index}>{text}</li>) //eslint-disable-line
  // If list exists, render the component
  if (list) {
    return (
      <ListContainer width={width}>
        {heading ? <h4>{heading}</h4> : null}
        <StyledList>{renderList(list)}</StyledList>
      </ListContainer>
    )
  }
  return null
}

List.propTypes = {
  heading: PropTypes.string,
  list: PropTypes.array,
  width: PropTypes.string,
}
export { List }
