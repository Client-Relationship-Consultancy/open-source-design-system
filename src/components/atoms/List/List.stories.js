/* eslint no-param-reassign: 0 */

import React from "react"

import { storiesOf } from "@storybook/react"
import { select } from "@storybook/addon-knobs"

import List from "./List"
import Heading from "../Heading"
import Paragraph from "../Paragraph"

const stories = storiesOf("List", module)

stories.add("Basic List", () => {
  const options = ["ul", "ol"]

  return (
    <>
      <Heading level={3}>A list</Heading>
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
        laboris nisi ut aliquip ex ea commodo consequat.
      </Paragraph>
      <List listType={select("List type", options, "ul")}>
        <li>List 1</li>
        <li>List 2</li>
        <li>List 3</li>
      </List>
      <Paragraph lastParagraph>
        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
        pariatur. Excepteur sint occaecat cupidatat non proident.
      </Paragraph>
    </>
  )
})
