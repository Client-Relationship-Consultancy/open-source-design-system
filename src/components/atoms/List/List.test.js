import React from "react"
import { shallow } from "enzyme"
import List from "./List"
import "jest-styled-components"

describe("List", () => {
  it("should render snapshot", () => {
    const component = shallow(
      <List listType="ul">
        <li>List 1</li>
        <li>List 2</li>
      </List>,
    )
    expect(component).toMatchSnapshot()
  })

  it("should render a ul when no listType is given", () => {
    const component = shallow(<List />)
    expect(component.find("StyledList").props().as).toEqual("ul")
  })

  it("should render a ol when listType is ol", () => {
    const component = shallow(<List listType="ol" />)
    expect(component.find("StyledList").props().as).toEqual("ol")
  })

  it("should render a ul when listType is ul", () => {
    const component = shallow(<List listType="ul" />)
    expect(component.find("StyledList").props().as).toEqual("ul")
  })
})
