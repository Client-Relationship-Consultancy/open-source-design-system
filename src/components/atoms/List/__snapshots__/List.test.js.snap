// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render snapshot 1`] = `
<StyledList
  as="ul"
>
  <li>
    List 1
  </li>
  <li>
    List 2
  </li>
</StyledList>
`;
